tests/integ/datasource/test_postgres.py

```diff
@@ -0,0 +1,486 @@
+#
+# Copyright (c) 2012-2025 Snowflake Computing Inc. All rights reserved.
+#
+
+import datetime
+from decimal import Decimal
+
+from snowflake.snowpark import Row
+from snowflake.snowpark._internal.data_source.drivers import Psycopg2Driver
+from snowflake.snowpark._internal.data_source.drivers.psycopg2_driver import (
+    Psycopg2TypeCode,
+)
+from snowflake.snowpark._internal.data_source.utils import DBMS_TYPE
+from snowflake.snowpark.exceptions import SnowparkDataframeReaderException
+from snowflake.snowpark.types import (
+    StructType,
+    StructField,
+    StringType,
+    DecimalType,
+    DoubleType,
+    LongType,
+    BinaryType,
+    BooleanType,
+    VariantType,
+    DateType,
+    TimestampType,
+    TimeType,
+    TimestampTimeZone,
+)
+from tests.parameters import POSTGRES_CONNECTION_PARAMETERS
+from tests.utils import IS_IN_STORED_PROC
+import pytest
+
+DEPENDENCIES_PACKAGE_UNAVAILABLE = True
+try:
+    import psycopg2  # noqa: F401
+    import pandas  # noqa: F401
+
+    DEPENDENCIES_PACKAGE_UNAVAILABLE = False
+except ImportError:
+    pass
+
+pytestmark = [
+    pytest.mark.skipif(DEPENDENCIES_PACKAGE_UNAVAILABLE, reason="Missing 'psycopg2'"),
+    pytest.mark.skipif(IS_IN_STORED_PROC, reason="Need External Access Integration"),
+]
+
+
+POSTGRES_TABLE_NAME = "test_schema.ALL_TYPE_TABLE"
+EXPECTED_TEST_DATA = [
```

Comment:
I suggest we separate the data with test itself, we could do following structure:
-test/integ
--data_source
---test_postgres
-test/resource
--test_data_source_dir
---test_postgres_data.py